WalletWasabi/Services/WalletService.cs

```diff
@@ -632,17 +633,11 @@ private async Task<bool> ProcessTransactionAsync(SmartTransaction tx)
 					walletRelevant = true;
 
 					foundKey.SetKeyState(KeyState.Used, KeyManager);
-					List<SmartCoin> spentOwnCoins = Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).ToList();
+					spentOwnCoins = spentOwnCoins ?? Coins.Where(x => tx.Transaction.Inputs.Any(y => y.PrevOut.Hash == x.TransactionId && y.PrevOut.N == x.Index)).ToList();
 					var anonset = tx.Transaction.GetAnonymitySet(i);
 					if (spentOwnCoins.Count != 0)
 					{
 						anonset += spentOwnCoins.Min(x => x.AnonymitySet) - 1; // Minus 1, because do not count own.
-
-						// Cleanup exposed links where the txo has been spent.
-						foreach (var input in spentOwnCoins.Select(x => x.GetTxoRef()))
-						{
-							ChaumianClient.ExposedLinks.TryRemove(input, out _);
```

Comment:
Move this from here where outputs are being processed to below where the inputs are processed.